duolingo.py

```diff
@@ -28,20 +28,84 @@ class DuolingoException(Exception):
 class Duolingo(object):
     USER_AGENT = "Python Duolingo API/{}".format(__version__)
 
-    def __init__(self, username, password):
-        self.username = username
-        self.password = password
-        self.user_url = "https://duolingo.com/users/%s" % self.username
+    _url_base = "https://{host}{base_path}"
+    _url_user_no_auth = "/users?username={username}"
+    _url_login = "/login?fields="
+
+    def __init__(self, username="", password="", host="www.duolingo.com", base_path="/2017-06-30"):
```

Comment:
I feel like it would be better to be keyword-only arguments for those last 2, like this?
`username="", password="", *, host="www.duolingo.com", base_path="/2017-06-30"`
Seems more clear, when folks are overriding the host and base path defaults. What do you think?

Reply:
My feeling is that username and password have to be removed from initialisation and be provided in the login method, so I made keyword arguments for the backward computability with the current implementation, so that there is no need to rewrite the whole module.

Reply:
I think we need to keep backward compatibility for users providing the username and password in the initialisation, but adding the `*, ` between them and the host and base_path arguments just makes it a bit more clear, forcing users to do:
`Duolingo("user", "pass", host="www.duolingo.com", base_path="/2017-06-30")`
rather than allowing
`Duolingo("user", "pass", "www.duolingo.com", "/2017-06-30")`
which seems a bit less clear to me
What do you think?